### ISSUE TITLE
rebase-helper encounters a commit_log object that is None and crashes

### ISSUE THREAD

**Issue by jeremycline** [rebase-helper/rebase-helper#249]
the-new-hotness reliably fails on Fedora 25 with the following:

```
[2016-12-14 16:21:06][rebase-helper    INFO] Git-rebase operation to cryptography-1.7.1 is ongoing...
[2016-12-14 16:21:06][hotness.consumers WARNING] rebase-helper failed with: 'NoneType' object has no attribute '__getitem__'
[2016-12-14 16:21:06][hotness.consumers WARNING] Traceback (most recent call last):
  File "/home/vagrant/devel/hotness/consumers.py", line 786, in _rebase
    result_rh, rh_stuff = self.buildsys.rebase(package, upstream, tmp)
  File "/home/vagrant/devel/hotness/buildsys.py", line 231, in rebase
    return self._run_rh(package, args, tmp_dir)
  File "/home/vagrant/devel/hotness/buildsys.py", line 209, in _run_rh
    result_rh = rh_app.run()
  File "/usr/lib/python2.7/site-packages/rebasehelper/application.py", line 711, in run
    self.patch_sources(sources)
  File "/usr/lib/python2.7/site-packages/rebasehelper/application.py", line 410, in patch_sources
    **self.kwargs)
  File "/usr/lib/python2.7/site-packages/rebasehelper/patch_helper.py", line 371, in patch
    return self._tool.run_patch(old_dir, new_dir, rest_sources, git_helper, patches, prep, **kwargs)
  File "/usr/lib/python2.7/site-packages/rebasehelper/patch_helper.py", line 331, in run_patch
    return cls._git_rebase()
  File "/usr/lib/python2.7/site-packages/rebasehelper/patch_helper.py", line 150, in _git_rebase
    init_hash, last_hash = cls._prepare_git(upstream)
  File "/usr/lib/python2.7/site-packages/rebasehelper/patch_helper.py", line 119, in _prepare_git
    last_hash = GitHelper.get_commit_hash_log(cls.output_data, number=number)
  File "/usr/lib/python2.7/site-packages/rebasehelper/utils.py", line 758, in get_commit_hash_log
    commit = commit_log[number]
TypeError: 'NoneType' object has no attribute '__getitem__'
[2016-12-14 16:21:09][hotness.bz    INFO] Followed up on bug: https://partner-bugzilla.redhat.com/show_bug.cgi?id=1392812
```

**Comment by nforro** [comment]
I'm not able to reproduce this on F25 rebasing `python-cryptography` from `1.5.3-5` to `1.7.1`.

Would it be possible for you to attach complete `rebase-helper-debug.log` file?
Even better would be if you could attach entire `rebase-helper-workspace` directory, as I'm afraid the log file wouldn't be enough.

Thanks.

**Comment by nforro** [comment]
Ok, reproduced using Vagrant environment. The problem is that `rebase-helper` is expecting valid git configuration, but there isn't one on freshly provisioned system.

Fixed in PR #250.
